src/include/catalog/index_key_schema.h

```diff
@@ -0,0 +1,95 @@
+#pragma once
+
+#include <vector>
+#include "catalog/catalog_defs.h"
+#include "type/type_id.h"
+
+namespace terrier::catalog {
+
+/**
+ * A column of the index key has an identifier, type, and describes whether it can be null.
+ */
+class IndexKeyColumn {
```

Comment:
How is this different than `IndexKeyDefinition`?

Reply:
It doesn’t take an OID in its constructor which is the same separation as the schema columns.  This allows the catalog to be the single point for assigning OIDs.